From c51ca46a847560961d7219b2535793b3f09eea92 Mon Sep 17 00:00:00 2001
From: Simon Mikuda <simon.mikuda@streamunlimited.com>
Date: Thu, 11 Apr 2019 15:53:44 +0200
Subject: [PATCH] hci: Don't send "LE Read Remote Used Features" request

Reply from read remote features is not actually used and it can cause
connection hang if encryption is requested before request completes.

Signed-off-by: Simon Mikuda <simon.mikuda@streamunlimited.com>
---
 net/bluetooth/hci_event.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index 2381546..8c3ad64 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -4565,8 +4565,13 @@ static void hci_le_conn_complete_evt(struct hci_dev *hdev, struct sk_buff *skb)
 		 * role is possible. Otherwise just transition into the
 		 * connected state without requesting the remote features.
 		 */
+		/* Reply from read remote features is not actually used
+		 * and it can cause connection hang if encryption is
+		 * requested before request completes.
 		if (conn->out ||
 		    (hdev->le_features[0] & HCI_LE_SLAVE_FEATURES)) {
+		*/
+		if (false) {
 			struct hci_cp_le_read_remote_features cp;
 
 			cp.handle = __cpu_to_le16(conn->handle);
-- 
2.7.4

